compiler/build.gradle

```diff
@@ -73,9 +73,46 @@ model {
   }
 }
 
+configurations {
+  testNanoCompile
+}
+
 dependencies {
   testCompile project(':grpc-protobuf'),
               project(':grpc-stub')
+  testNanoCompile project(':grpc-protobuf-nano'),
+                  project(':grpc-stub')
+}
+
+sourceSets {
+  test {
+    proto {
+      plugins {
+        grpc { }
+      }
+    }
+  }
+  testNano {
+    proto {
+      setSrcDirs(['src/test/proto'])
+      builtins {
+        remove java
+        javanano {
+          option 'ignore_services=true'
+        }
+      }
+      plugins {
+        grpc {
+          option 'nano=true'
+        }
+      }
+    }
+    compileClasspath += configurations.testNanoCompile
+  }
+}
+
+checkstyleTestNano {
+  source = fileTree(dir: "src/testNano", include: "**/*.java")
```

Comment:
checkstyle creates a checkstyleTestNano for the testNano sourceSet, which seems to include everything under the project, including ``build/generated-sources``. I just need to point the source to nothing, but I think it's nice to be consistent with other checkstyle tasks.